src/streams/openssl.c

```diff
@@ -602,6 +600,8 @@ int openssl_connect(git_stream *stream)
 	if ((ret = SSL_connect(st->ssl)) <= 0)
 		return ssl_set_error(st->ssl, ret);
 
+	st->connected = true;
```

Comment:
The same pattern exists in mbedtls code, so we might want to do the same check over there, too.

Reply:
This has been rebased out.